### ISSUE TITLE
Lose ability to jump

### ISSUE THREAD

**Issue by godofgrunts** [Aaron-SP/bds#6]
I think I've only noticed it on top of trees, but sometimes I lose the ability to jump until I fall off of the tree and land somewhere else.

If I were to guess, my character isn't register as landing.

**Comment by Aaron-SP** [comment]
Yup, It's been evading me for a while. I think it has something to do with when landing at a certain velocity that the land detection code misses a land and so you are still "falling". I can't reliably do it enough to debug, so it hasn't been fixed yet. We can leave this issue up until someone gets some inspiration.

**Comment by Aaron-SP** [comment]
I found the error. I updated this in MGL https://github.com/Aaron-SP/mgl/commit/982e79a90d20ec37b2c354fc2ed937553af2a672 and https://github.com/Aaron-SP/bds/commit/7330659fae5477b783ece695c6cc5401950666cb along with a bug I introduced in the last commit. The save file has been altered so don't try to load an old save file.

The problem was the jump counter never got reset if the player lands with a positive Y velocity. I reset the land counter after each jump which should guarantee the jump counter gets reset but we will see if I am correct.
